File: python3/27.remove-element.py
#
# @lc app=leetcode id=27 lang=python3
#
# [27] Remove Element
#

# @lc code=start
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        slow = 0
        for fast in range(0, len(nums)):
            if (val != nums[fast]):
                nums[slow] = nums[fast]
                slow += 1
        
# @lc code=end

File: typescript/27.remove-element.ts
/*
 * @lc app=leetcode id=27 lang=typescript
 *
 * [27] Remove Element
 */

// @lc code=start
function removeElement(nums: number[], val: number): number {
    let fast = 0, slow = 0;
    while (fast < nums.length) {
        // [1,1,1,2,2,3] 2
        //  f         f
        //  s     s
        if (nums[fast] === val) {
            fast += 1;
            continue;
        }
        nums[slow] = nums[fast];
        fast += 1;
        slow += 1;
    }

    return slow;
};
// @lc code=end

